credentialz/README.md

```diff
@@ -341,6 +341,126 @@ stream.Send(
 )
 ```
 
+## User Journeys
+
+### Rotate Certificate based on existing key
+
+The most common operation we are expecting to require on devices is the rotation of certificates used for SSH access for devices. This operation expects to reuse the existing host key on the device.
+
+* Get the public key configured on the host.
+
+```go
+
+resp, err := c.GetPublicKeys(&GetPublicKeyRequests{})
+```
+
+* Generate certificate based on key.
+
+* Rotate certificate on device.
+
+```go
+stream.Send(
+    RotateHostCredentialsRequest {
+        server_keys: ServerKeysRequest {
+            certificate: "A....=",
+            version: "v1.0",
+            created_on: 3214451134,
+        }
+    }
+)
+```
+
+* Validate that new settings are working as expected.
+
+* Finalize request.
+
+```go
+stream.Send(
+    RotateHostCredentialsResponse {
+        finalize: FinalizeRequest {}
+    }
+)
+```
+
+### Generate new host key and rotate Certificate based on the new key
+
+This use case focuses on the rotation of a host key and then generation of the certificate based on the new public key.
+
+* Send request for generation of new private key.
+
+```go
+stream.Send(
+    RotateHostCredentialsRequest {
+        generate_keys_request: GenerateKeysRequest {
```

Comment:
`generate_keys_request` has been updated to `generate_keys` in the current review, please update this accordingly.